src/rulesets/api-lifeycle.ts

```diff
@@ -35,4 +35,45 @@ export const rules = {
       },
     );
   },
+  followSunsetRules: ({ checkApiContext }: SnykApiCheckDsl) => {
+    checkApiContext.must("follow sunset rules", (context) => {
+      if (!context.wasDeleted) return;
+      const deprecatedBy =
+        context.resourceVersions?.[context.changeResource]?.[
+          context.changeVersion.date
+        ]?.[context.changeVersion.stability]?.deprecatedBy;
+
+      // This means the resource was never deprecated
+      if (!deprecatedBy) {
+        expect.fail(
+          `expected ${context.changeResource} to be deprecated before removing`,
+        );
+        return;
+      }
+
+      const { stability } = context.changeVersion;
+      const resourceName = context.changeResource;
+
+      // Make sure the sunset schedule is followed now
+      const contextDate = new Date(context.changeDate);
+      const resourceDate = new Date(context.changeVersion.date);
+      const diffDays =
+        (contextDate.getTime() - resourceDate.getTime()) /
+        (1000 * 60 * 60 * 24);
+
+      // Number of days required before a resource can be removed
+      const sunsetSchedule = {
+        experimental: 30,
+        beta: 90,
+        ga: 180,
+      };
+      const requiredDays = sunsetSchedule[stability];
```

Comment:
This rule should ignore `wip` stability.

Reply:
Ah, the entire rule, correct? So you can delete a wip even if it's not deprecated?

Reply:
Yes, that is fine. `wip` releases are never published.

Reply:
I think this still needs to be dealt with. If we land as-is, we won't be able to remove `wip` versions? Or the behavior will be undefined (what does "(number) < null" evaluate to in JS?)